src/middleware.ts

```diff
@@ -0,0 +1,54 @@
+import { NextRequest, NextResponse } from "next/server";
+import {
+  ERROR_INVALID_SESSION_SCHEMA,
+  SESSION_COOKIE_NAME,
+  verifySession,
+} from "./lib/tools/session";
+
+export async function middleware(request: NextRequest) {
+  const sessionCookie = request.cookies.get(SESSION_COOKIE_NAME)?.value;
```

Comment:
Can we not use the `cookies()` function here?

Reply:
Yeah, we can. I'll make a new commit. I thought I couldn't earlier for some reason.

Reply:
Oh, yep. It's not applicable for the response. We need to modify the response's cookies and for some reason, `clearSessionCookie()` doesn't work for the middleware.

Reply:
The get() function works, so I'm able to clear 3 lines of code.